system/index.js

```diff
@@ -29,18 +29,21 @@ const userFn = getFunction(USER_FUNCTION_URI);
 module.exports = async ({headers, payload}) => {
   if (DEBUG) {
     console.log('==System Function Start==');
-    console.log(`HEADERS: ${JSON.stringify(headers)}`);
+    console.log(`HEADERS: ${headers}`);
     console.log(`ORIGINAL PAYLOAD: ${payload}`);
     console.log(`MIDDLEWARE_FUNCTION_URI: ${MIDDLEWARE_FUNCTION_URI}`);
     console.log('==Middleware Function(s) Start==');
   }
 
+  const req = { payload };
+  let res = [];
+
   await Promise.all(middlewareFns.map(async (middleware) => {
         try {
           if (DEBUG) {
-            console.log(`MIDDLEWARE PAYLOAD: ${payload}`);
+            console.log(`MIDDLEWARE REQ: ${req}`);
           }
-          payload = await middleware(payload);
+          res = await middleware(req);
```

Comment:
This looks like each middleware would end up overriding the previous middleware's result. Should this be appending to a final result?

Reply:
Yeah, I need to fix that.  Good catch 👍